blueprints/flexible-topologies.md

```diff
@@ -250,11 +250,29 @@ response JSON:
 
 ##### `/deliveryservices` endpoints
 
-All relevant Delivery Service APIs will have their JSON request and response objects modified to include a new `topology` field which references the name of the topology it's assigned to:
-```JSON
+All relevant Delivery Service APIs will have their JSON request and response objects modified to include the following new fields:
+- `topology` - the name of the topology it's assigned to
+- `firstHeaderRewrite` - the header_rewrite ATS config that will be applied to the *first tier* caches in the Delivery Service's Topology
+- `middleHeaderRewrite` - the header_rewrite ATS config that will be applied to all *middle tier* caches in this Delivery Service's Topology
+- `lastHeaderRewrite` - the header_rewrite ATS config that will be applied to all *last tier* caches in this Delivery Service's Topology
```

Comment:
IMO `midHeaderRewrite` sounds too close to the existing `midHeaderRewrite` rules that we have today.

It would be nice if `firstHeaderRewrite`/`lastHeaderRewrite` were named something that implies their place in the topology, like `leafHeaderRewrite`/`rootNodeHeaderRewrite`.

Reply:
Yeah, I was considering `intermediateHeaderRewrite`/`innerHeaderRewrite` vs `middleHeaderRewrite` because it is close like you said. Honestly I would like to continue using `edgeHeaderRewrite` too since that still makes sense in the topology world, but I'm not sure we can take those fields as-is without some manual migration to the new fields. `root` can be confusing because origins are really the "root", but they don't have to be in the topology. Plus a topology isn't truly a tree, so there isn't really a root either.

Reply:
Edge/Parents/LastParent ?

Reply:
Feel free to resolve this convo since your response makes sense and @jhg03a additionally suggested Edge/Parents/LastParent elsewhere. I don't have permission to or I would.